src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -874,7 +874,7 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 					}
 				} else {
 					const sessionId = isResponseVM(element) || isRequestVM(element) ? element.sessionId : '';
-					const blockModel = this.codeBlockModelCollection.get(sessionId, element.id, index);
+					const blockModel = this.codeBlockModelCollection.getOrCreate(sessionId, element.id, index);
```

Comment:
@mjbvz said to try something like this for our case

https://github.com/microsoft/vscode/blob/957ccc60506e170450794caceca70f560f4d14cc/src/vs/workbench/contrib/chat/common/chatViewModel.ts#L269